dbt_meshify/dbt_projects.py

```diff
@@ -72,22 +75,27 @@ def __init__(
         )
 
     def _get_xproj_children_of_selected_nodes(self) -> Set[str]:
-        return {
-            model.unique_id
-            for model in self.models.values()
-            if any(parent for parent in model.depends_on.nodes if parent in self.resources)
-            and model.unique_id not in self.resources
-        }
+        all_children: Set[str] = set()
+        for resource in self.resources:
+            if resource.startswith("test"):
+                continue
+            all_children.update(self.child_map.get(resource, {}))
+        return all_children - self.resources
 
     def _get_xproj_parents_of_selected_nodes(self) -> Set[str]:
-        resources = [self.get_manifest_node(resource) for resource in self.resources]
-        return {
-            node
-            for resource in resources
-            if resource is not None and isinstance(resource, (ModelNode, SnapshotNode))
-            for node in resource.depends_on.nodes
-            if node not in self.resources
-        }
+        all_parents: Set[str] = set()
+        for resource in self.resources:
+            if resource.startswith("test"):
+                continue
+            all_parents.update(self.parent_map.get(resource, {}))
+        return all_parents - self.resources
+
+    def _build_parent_and_child_maps(self) -> None:
+        if hasattr(self.manifest, "child_map") and hasattr(self, "parent_map"):
+            return
```

Comment:
I'm not sure that I understand what this logic is here for. Can I get a quick explanation?

Reply:
this logic looks a little drunk tbh -- let me double check whether this was a type check thing that may be eschewed altogether

Reply:
I removed the `self.child_map = {}` and `self.parent_map = {}` from the `init` method, and removed this. IIRC i was battling with mypy when i was writing this manifest method in a slightly different order. I think this should be sufficient!